static/js/hashchange.js

```diff
@@ -307,6 +307,10 @@ exports.get_current_hash = function () {
     return window.location.hash;
 };
 
+exports.test_regex_in_current_hash = function (regex) {
+    return regex.test(window.location.hash);
+}
```

Comment:
Similarly, what I'd had in mind here is replacing the check in `update_default_streams_table` with a check for what page we're looking at... realistically, that shouldn't use `hashchange` details at all, but instead look at cleaner state for whether that page is currently loaded (I think this is done via `meta.loaded` in some other settings pages).

I'm actually not sure that the `update_default_streams_table` logic is even checking for the page where the default streams table is present correctly -- that code might just be wrong in that way too.

Reply:
Or if we want a generic function here, I think e.g. hash_util.is_page("streams/new") would be a better interface.  Part of what I had in mind is to end the practice of having random regexes scattered around the codebase looking at the URL to determine what's happening.